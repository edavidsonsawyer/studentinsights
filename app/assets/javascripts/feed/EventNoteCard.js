import PropTypes from 'prop-types';
import React from 'react';
import {eventNoteTypeText} from '../helpers/eventNoteType';
import Educator from '../components/Educator';
import NoteText from '../components/NoteText';
import HouseBadge from '../components/HouseBadge';
import NoteBadge from '../components/NoteBadge';
import Timestamp from '../components/Timestamp';
import FeedCardFrame from './FeedCardFrame';


// Render a card in the feed for an EventNote
export default class EventNoteCard extends React.Component {
  render() {
    const {eventNoteCardJson, iconsEl, style, children} = this.props;
    const {student, educator} = eventNoteCardJson;

    return (
      <div className="EventNoteCard">
        <FeedCardFrame
          style={style}
          student={student}
          byEl={
            <div>
              <span>by </span>
              <Educator
                style={styles.person}
                educator={educator} />
            </div>
          }
          whereEl={<div>in {eventNoteTypeText(eventNoteCardJson.event_note_type_id)}</div>}
          whenEl={<Timestamp railsTimestamp={eventNoteCardJson.recorded_at} />}
          badgesEl={<div>
            {student.house && <HouseBadge style={styles.footerBadge} house={student.house} />}
            <NoteBadge style={styles.footerBadge} eventNoteTypeId={eventNoteCardJson.event_note_type_id} />
          </div>}
          iconsEl={iconsEl}
        >
          {children || <NoteText text={eventNoteCardJson.text} />}
        </FeedCardFrame>
      </div>
    );
  }
}
EventNoteCard.propTypes = {
  eventNoteCardJson: PropTypes.shape({
    id: PropTypes.number.isRequired,
    recorded_at: PropTypes.string.isRequired,
    event_note_type_id: PropTypes.number.isRequired,
    text: PropTypes.string.isRequired,
    educator: PropTypes.object.isRequired,
    student: PropTypes.shape({
      id: PropTypes.number.isRequired,
      first_name: PropTypes.string.isRequired,
      last_name: PropTypes.string.isRequired,
      grade: PropTypes.string.isRequired,
      house: PropTypes.string,
      school: PropTypes.shape({
        local_id: PropTypes.string.isRequired,
        school_type: PropTypes.string.isRequired
      }),
      homeroom: PropTypes.shape({
        id: PropTypes.number.isRequired,
        name: PropTypes.string.isRequired,
        educator: PropTypes.object
      })
    })
  }).isRequired,
  iconsEl: PropTypes.node,
  style: PropTypes.object,
  children: PropTypes.node
};


const styles = {
  footerBadge: {
    marginLeft: 5
  },
  person: {
    fontWeight: 'bold'
  }
};
