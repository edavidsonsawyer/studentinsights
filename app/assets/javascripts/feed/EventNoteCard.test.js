import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import EventNoteCard from './EventNoteCard';
import {withDefaultNowContext} from '../testing/NowContainer';
import PerDistrictContainer from '../components/PerDistrictContainer';

function testProps(props = {}) {
  return {
    eventNoteCardJson: {
      id: 321,
      recorded_at: '2004-03-05T00:00:00.000Z',
      event_note_type_id: 305,
      text: 'hello!',
      educator: {
        id: 4,
        email: 'kt@demo.studentinsights.org',
        full_name: 'Teacher, Kevin',
      },
      student: {
        id: 55,
        first_name: 'Mari',
        last_name: 'Skywalker',
        grade: '9',
        house: 'Beacon',
        has_photo: true,
        school: {
          local_id: 'SHS',
          school_type: 'HS'
        },
        homeroom: {
          id: 13,
          name: 'SHS-052',
          educator: {
            id: 5,
            email: 'lt@demo.studentinsights.org',
            full_name: 'Teacher, Lois',
          }
        }
      }
    },
    ...props
  };
}

function testEl(props = {}, context = {}) {
  const districtKey = context.districtKey || 'somerville';
  return withDefaultNowContext(
    <PerDistrictContainer districtKey={districtKey}>
      <EventNoteCard {...props} />
    </PerDistrictContainer>
  );
}

it('renders without crashing', () => {
  const props = testProps();
  const el = document.createElement('div');
  ReactDOM.render(testEl(props), el);
});

it('matches snapshot', () => {
  const props = testProps();
  const tree = renderer
    .create(testEl(props))
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('snaphots when passing alternate children', () => {
  const props = testProps({children: <b>hello!</b>});
  const tree = renderer
    .create(testEl(props))
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders when student has no homeroom', () => {
  const props = testProps();
  delete props.eventNoteCardJson.student.homeroom;
  const el = document.createElement('div');
  ReactDOM.render(testEl(props), el);
});
