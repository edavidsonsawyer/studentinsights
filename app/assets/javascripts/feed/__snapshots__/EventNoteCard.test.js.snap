// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="EventNoteCard"
>
  <div
    className="Card FeedCardFrame"
    style={
      Object {
        "border": "1px solid #ccc",
        "borderRadius": 3,
        "padding": 10,
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "flex-start",
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "flex-start",
            }
          }
        >
          <div>
            <a
              href="/students/55"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Mari
               
              Skywalker
            </a>
          </div>
          <div>
            9th grade
          </div>
          <div />
        </div>
      </div>
      <div
        style={
          Object {
            "alignItems": "flex-end",
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <div>
          <div>
            <span>
              by 
            </span>
            <a
              className="Educator"
              href="mailto:kt@demo.studentinsights.org"
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Kevin Teacher
            </a>
          </div>
        </div>
        <div>
          <div>
            in 
            9th Grade Experience
          </div>
        </div>
        <div>
          <div
            className="Timestamp"
            style={Object {}}
          >
            14 years ago
             on 
            3/5
          </div>
        </div>
      </div>
    </div>
    <div
      style={
        Object {
          "marginBottom": 20,
          "marginTop": 20,
        }
      }
    >
      <div
        className="NoteText"
        style={
          Object {
            "border": "1px solid transparent",
            "fontFamily": "'Open Sans', sans-serif",
            "fontSize": 14,
            "marginTop": 5,
            "overflowX": "hidden",
            "padding": 0,
            "whiteSpace": "pre-wrap",
            "wordBreak": "break-word",
          }
        }
      >
        hello!
      </div>
    </div>
    <div
      className="FeedCardFrame-footer"
      style={
        Object {
          "display": "flex",
          "justifyContent": "space-between",
          "marginBottom": 5,
        }
      }
    >
      <div />
      <div>
        <span
          className="HouseBadge"
          style={
            Object {
              "backgroundColor": "#333",
              "color": "white",
              "marginLeft": 5,
              "opacity": 0.5,
              "padding": 5,
            }
          }
        >
          Beacon house
        </span>
        <span
          className="Badge"
          style={
            Object {
              "backgroundColor": "#139DEA",
              "color": "white",
              "marginLeft": 5,
              "opacity": 0.5,
              "padding": 5,
            }
          }
        >
          9th Grade Experience
        </span>
      </div>
    </div>
  </div>
</div>
`;
